cmd/crc/cmd/config/unset.go

```diff
@@ -21,6 +22,10 @@ func configUnsetCmd(config config.Storage) *cobra.Command {
 			if err != nil {
 				return err
 			}
+
+			properties := telemetry.PropertiesFromContext(cmd.Context())
+			properties.Set("key", args[0])
```

Comment:
I'd prefer to have a `config.Watch()` method which allows to register a watcher implementing a given interface, and then to only refer to this interface in the `config` code rather than hardcoding telemetry there.

Reply:
`config.Watch` pattern works in the daemon perspective. In `crc config set`, the watcher doesn't exist and if it was, it will likely never see the event as the CLI will quit as soon as possible.

Reply:
Imo `config.Watch(configWatcherForTelemetry)` can be called somewhere in `cmd/` to avoid having a hard dependency between config and telemetry.

Reply:
Then, how do you ensure when the CLI quits, that config.Watch correctly saw the event ? You need an extra channel and wait for it.

```
func init() {
  config.Watch(func() {
    // do stuff with the event
    done <- struct{}{}
  })
}

func configSet() {
  config.Set(key, value)
  <-done
}
```

`<-done` cannot be added in `postRun()` since we don't know if an event will be sent. 

This is not better at the end.

Reply:
Never mind, ignore this, when I made that comment I thought there was an added dependency on the `telemetry` package in `pkg/crc/config`. But these changes are all in `cmd/` where it's going to be less of a concern.